packages/gatsby/src/redux/nodes.ts

```diff
@@ -201,7 +208,7 @@ const ensureIndexByTypedChain = (
       node.__gatsby_resolved = resolvedNodes?.get(node.id)
     }
 
-    let v = node
+    let v = node as any // TODO: This might need refactoring?
```

Comment:
Can you elaborate?

This considers the object as a tree and basically walks the tree while confirming each vertex only has one edge (each obj has one prop). It doesn't care about the actual structure, it's not relevant and could be anything. Not sure how you would type this meta information but also not sure how you could do this dynamic bit within the realms of typing.

If at all concerning, you could tuck this in an explicitly uncovered function that accepts an arbitrary object and returns a boolean. You can probably not blackbox it and I don't know if TS supports saying "this param can be any object".

Good example of potential burdens TS introduce for us.

Reply:
i removed the TODO. This code is legit, I put it there while I was still learning what it did. 👍